packages/eth/src/delegate-manager/delegateManager.ts

```diff
@@ -0,0 +1,409 @@
+      functionName: 'getTotalLockedDelegationForServiceProvider',
+      args: [serviceProviderAddress]
+    })
+
+  getTotalDelegatorStake = ({
+    delegatorAddress
+  }: {
+    delegatorAddress: `0x${string}`
+  }) =>
+    this.client.readContract({
+      address: this.address,
+      abi,
+      functionName: 'getTotalDelegatorStake',
+      args: [delegatorAddress]
+    })
+
+  getDelegatorStakeForServiceProvider = ({
+    delegatorAddress,
+    serviceProviderAddress
+  }: {
+    delegatorAddress: `0x${string}`
+    serviceProviderAddress: `0x${string}`
+  }) =>
+    this.client.readContract({
+      address: this.address,
+      abi,
+      functionName: 'getDelegatorStakeForServiceProvider',
+      args: [delegatorAddress, serviceProviderAddress]
+    })
+
+  getPendingRemoveDelegatorRequest = ({
+    delegatorAddress,
+    serviceProviderAddress
+  }: {
+    delegatorAddress: `0x${string}`
+    serviceProviderAddress: `0x${string}`
+  }) =>
+    this.client.readContract({
+      address: this.address,
+      abi,
+      functionName: 'getPendingRemoveDelegatorRequest',
+      args: [serviceProviderAddress, delegatorAddress]
+    })
+
+  getPendingUndelegateRequest = ({
+    delegatorAddress
+  }: {
+    delegatorAddress: `0x${string}`
+  }) =>
+    this.client.readContract({
+      address: this.address,
+      abi,
+      functionName: 'getPendingUndelegateRequest',
+      args: [delegatorAddress]
+    })
+
+  getUndelegateLockupDuration = () =>
+    this.client.readContract({
+      address: this.address,
+      abi,
+      functionName: 'getUndelegateLockupDuration'
+    })
+
+  getMaxDelegators = () =>
+    this.client.readContract({
+      address: this.address,
+      abi,
+      functionName: 'getMaxDelegators'
+    })
+
+  getMinDelegationAmount = () =>
+    this.client.readContract({
+      address: this.address,
+      abi,
+      functionName: 'getMinDelegationAmount'
+    })
+
+  getRemoveDelegatorLockupDuration = () =>
+    this.client.readContract({
+      address: this.address,
+      abi,
+      functionName: 'getRemoveDelegatorLockupDuration'
+    })
+
+  getRemoveDelegatorEvalDuration = () =>
+    this.client.readContract({
+      address: this.address,
+      abi,
+      functionName: 'getRemoveDelegatorEvalDuration'
+    })
+
+  getGovernanceAddress = () =>
+    this.client.readContract({
+      address: this.address,
+      abi,
+      functionName: 'getGovernanceAddress'
+    })
+
+  getServiceProviderFactoryAddress = () =>
+    this.client.readContract({
+      address: this.address,
+      abi,
+      functionName: 'getServiceProviderFactoryAddress'
+    })
+
+  getClaimsManagerAddress = () =>
+    this.client.readContract({
+      address: this.address,
+      abi,
+      functionName: 'getClaimsManagerAddress'
+    })
+
+  getStakingAddress = () =>
+    this.client.readContract({
+      address: this.address,
+      abi,
+      functionName: 'getStakingAddress'
+    })
+
+  getSPMinDelegationAmount = ({
```

Comment:
Is all of this generated somehow or did you have to write all these by hand? 😅

Reply:
i did write them by hand. we could export the auto generated stuff by default, but it's a bit less descriptive. they're all copy pasta though, so not too bad. curious for other takes

Reply:
I prefer descriptive. If it's not changing that often, the overhead of automating it probably isn't worth it?